### PR TITLE
docs: Document new TLS functionality in the Helm chart

### LINKED ISSUE #1639: TLS updates to Helm chart
### Describe the Issue

The Helm chart now supports using custom CAs with or without cert-manager.

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/kubernetes/security/kubernetes-tls/

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: June 2

Fixes #1639

- Added instructions with/without cert-manager.
- Added more troubleshooting details.
- Added an example of configuring listeners with different certificates.

Preview: https://deploy-preview-1645--redpanda-documentation.netlify.app/docs/manage/kubernetes/security/kubernetes-tls/

### CODE REVIEW

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 1]
```diff
@@ -116,11 +122,11 @@ ID    HOST                                             PORT
 
 </details>
 
-### Test external connections
+#### Test external connections
 
-The SAN list of your self-signed certificate does not include the IP addresses of your worker nodes. As a result, to test external connections, you must enable external access using a custom domain. When you enable external access using a custom domain, the custom domain is included in the SAN list. Then, you can use rpk on your local machine to communicate with the cluster externally using the self-signed certificate for encryption. 
+To test external connections, you need to enable external access using a custom domain. The SAN list of your self-signed certificate does not contain the IP addresses of your worker nodes, but when you enable external access using a custom domain, that domain is included in the SAN list. Then, you can use rpk on your local machine to communicate with the cluster externally using the self-signed certificate for encryption.
```
Comment (Feediver1) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:127]:
```suggestion
To test external connections, you must enable external access using a custom domain. The SAN list of your self-signed certificate does not contain the IP addresses of your worker nodes, but when you enable external access using a custom domain, that domain is included in the SAN list. Then, you can use rpk on your local machine to communicate with the cluster externally using the self-signed certificate for encryption.
```

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 2]
```diff
@@ -162,6 +168,284 @@ The SAN list of your self-signed certificate does not include the IP addresses o
...
+    - To create a Secret that includes your `tls.crt` and `tls.key` files:
+
+      ```bash
+      kubectl create secret tls <secret-name> \
+        --cert=<path>tls.crt \
+        --key=<path>tls.key \
+        -n redpanda
+      ```
+
+    - To include the `ca.crt` file, you need to create a generic secret:
```
Comment (Feediver1) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:300]:
```suggestion
    - To include the `ca.crt` file, you must create a generic secret:
```

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 3]
```diff
@@ -162,6 +168,284 @@ The SAN list of your self-signed certificate does not include the IP addresses o
...
+
+```bash
+rpk cluster info --brokers <subdomain>.<custom-domain>:<external-port>
+```
+
+## Use Secrets to manage TLS certificates without cert-manager
+
+If you prefer not to use cert-manager for TLS certificate management, you can manage your TLS configurations using Kubernetes Secrets. The following steps outline how to configure TLS with Secrets:
+
+1. Create a Kubernetes Secret that contains your TLS certificates.
```
Comment (Feediver1) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:287]: Wondering--why is "Secret" capitalized? When I wrote about this at Confluent, we always used lowercase: https://docs.confluent.io/platform/current/security/secrets.html
Unless it is a product or licensed feature, I'd say use lowercase. We don't have specific guidance on this in our Style Guide, but I'm thinking we should.

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 4]
```diff
@@ -162,6 +168,284 @@ The SAN list of your self-signed certificate does not include the IP addresses o
...
+
+```bash
+rpk cluster info \
+  --brokers <subdomain>.<custom-domain>:<external-kafka-api-port> \
+  --tls-truststore <path-to-ca-certificate>
+```
+
+## Use multiple certificates for different listeners
+
+The Redpanda cluster provides granular control over the TLS certificates used by different listeners. This level of flexibility allows you ensure the required level of security for each listener. For example, you can use a self-signed certificate for the internal RPC listener, while using a Certificate Authority (CA) certificate for other listeners such as the Kafka API.
```
Comment (Feediver1) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:404]:
```suggestion
The Redpanda cluster provides granular control over the TLS certificates used by different listeners. This level of flexibility enables you to ensure the required level of security for each listener. For example, you can use a self-signed certificate for the internal RPC listener, while using a Certificate Authority (CA) certificate for other listeners such as the Kafka API.
```

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 5]
```diff
@@ -162,6 +168,284 @@ The SAN list of your self-signed certificate does not include the IP addresses o
...
+
+By default, all listeners in Redpanda use the self-signed certificate defined globally in the `tls.certs.default` configuration. To customize the certificates for each listener, you can edit the `listeners.<listener-name>.tls.cert` setting
```
Comment (Feediver1) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:406]:
```suggestion
By default, all listeners in Redpanda use the self-signed certificate defined globally in the `tls.certs.default` configuration. To customize the certificates for each listener, you can edit the `listeners.<listener-name>.tls.cert` setting.
```

**Review by Feediver1 — APPROVED** [review]
Please implement edits here. Thx.

File: docs/manage/kubernetes/security/kubernetes-tls.mdx (hunk 3 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/security/kubernetes-tls.mdx:287]: It's a convention in Kubernetes to capitalize resource names. We call this out here: https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-production-deployment/#documentation-conventions

File: docs/manage/kubernetes/security/kubernetes-tls.mdx [hunk 6]
```diff
@@ -162,6 +168,288 @@ The SAN list of your self-signed certificate does not include the IP addresses o
     --tls-truststore ca.crt
   ```
 
+### Use a CA certificate
```
Comment (joejulian) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:171]: Technically, the self-signed also uses a CA certificate, it's just a self-signed one. The generated CA can be downloaded and used by external clients to validate their connections.

**Review by joejulian — APPROVED** [review]

File: docs/manage/kubernetes/security/kubernetes-tls.mdx (hunk 6 above)
Comment (JakeSCahill) [inline docs/manage/kubernetes/security/kubernetes-tls.mdx:171]:
```suggestion
### Use a public CA certificate
```

**Pull request merged**